packages/core/frontend/sirius-components-core/src/workbench/Workbench.tsx

```diff
@@ -97,35 +98,39 @@ export const Workbench = ({
 
   const { data: representationFactories } = useData(representationFactoryExtensionPoint);
 
-  const { error } = useSubscription<GQLEditingContextEventSubscription>(editingContextEventSubscription, {
+  const onData = ({ data }: OnDataOptions<GQLEditingContextEventSubscription>) => {
+    flushSync(() => {
+      const handleDataEvent: HandleSubscriptionResultEvent = {
+        type: 'HANDLE_SUBSCRIPTION_RESULT',
+        result: data,
+      };
+      dispatch(handleDataEvent);
+    });
+  };
+
+  const onError = ({ message }: ApolloError) => {
+    const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message };
+    dispatch(showToastEvent);
+  };
+
+  const onComplete = () => {
+    const completeEvent: HandleCompleteEvent = { type: 'HANDLE_COMPLETE' };
+    dispatch(completeEvent);
+  };
```

Comment:
I'm not familiar with `flushSync`, but if we need to use it with `onData`, shouldn't we also use it for `onError` and `onComplete`?

Reply:
I don't think so, we're using `flushSynch `here in order to force react to not batch `onData `events (in case there are multiple events).
In the case of `onError `and `onComplete `we will receive one event.

Reply:
Yes it's specifically for the restoration of the previous behavior with onData. See

- https://github.com/reactwg/react-18/discussions/21
- https://github.com/apollographql/apollo-client/issues/10001